### PR TITLE
Add input validation

### PR DESCRIPTION
**Pull request by saikiranakula-amzn**
*Description of changes:*
Added input validation
1. Added checks on the input field for the gRPC api to check for invalid characters
2. Check to validate credspec parsing

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: api/src/gmsa_service.cpp [hunk 1]
```diff
@@ -2151,21 +2203,28 @@ int parse_cred_spec_domainless( std::string credspec_data, creds_fetcher::krb_ti
...
+             contains_invalid_characters_in_ad_account_name(service_account_name))
         {
             std::cout << getCurrentTime() << '\t' << "ERROR: credentialspec file is not formatted"
                                                      " properly" <<
                 std::endl;
             return -1;
         }
 
+        // get credentialspec arn
+        std::string domainless_user_arn = root["ActiveDirectoryConfig"]["HostAccountConfig"]["PluginInput"]["CredentialArn"].asString();
```
Comment (as14692) [inline api/src/gmsa_service.cpp:2216]: Aren't "ActiveDirectoryConfig" and "HostAccountConfig" in the same level in the domainless json file?

**Review by as14692 — APPROVED** [review]

**Comment by as14692** [comment]
> _Issue #, if available:_
> 
> _Description of changes:_ Add input validation
> 
> By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

Please update the description

File: api/src/gmsa_service.cpp (hunk 1 above)
Comment (saikiranakula-amzn) [reply api/src/gmsa_service.cpp:2216]: we are using correct one - https://learn.microsoft.com/en-us/virtualization/windowscontainers/manage-containers/manage-serviceaccounts

Comment (as14692) [reply api/src/gmsa_service.cpp:2216]: Verified with the correct domainless json. Resolving

Comment (as14692) [reply api/src/gmsa_service.cpp:2216]: Add check for errors when accessing the JSON elements

**Pull request merged**